sound/soc/intel/boards/bxt_pcm512x.c

```diff
@@ -53,6 +54,15 @@ static int broxton_hdmi_init(struct snd_soc_pcm_runtime *rtd)
 
 	list_add_tail(&pcm->head, &ctx->hdmi_pcm_list);
 
+	/*
+	 * Setup the device_link to make sure that hdmi codec power is
+	 * turned on before the consumer starts to operate on the
+	 * codec register.
+	 */
+	if (!ctx->link)
+		ctx->link = device_link_add(rtd->card->dev, dai->dev,
```

Comment:
@libinyang looks like we need to check for null after device_link_add as it could return null?

Reply:
If it returns NULL, it means the link is not added. But I don't think we should stop the init. Or we add a dev_warn?

Reply:
hmm a warning will be good I think